waldo/upload.go

```diff
@@ -190,58 +268,124 @@ func (ua *UploadAction) makeArgs(buildPath, uploadToken string) []string {
 		args = append(args, "--verbose")
 	}
 
-	if ua.options.Version {
-		args = append(args, "--version")
-	}
-
 	return args
 }
 
-func (ua *UploadAction) mungeArgs(ud *data.UserData, recipe *data.Recipe) ([]string, string, error) {
-	uploadToken := ua.options.UploadToken
+func (ua *UploadAction) processOptions(ciMode bool) (*data.UserData, *data.Recipe, error) {
+	isRecipe := ua.isRecipeTarget()
 
-	// if len(uploadToken) == 0 {
-	// 	uploadToken = recipe.UploadToken
-	// }
+	var err error
 
-	if len(uploadToken) == 0 {
-		uploadToken = os.Getenv("WALDO_UPLOAD_TOKEN")
+	//
+	// - An app token (formerly known as an “upload token”) is _allowed_
+	//   only if the target is _not_ a recipe name.
+	// - An app token is _required_ if and only if we are in CI Mode.
+	//
+	ua.appToken, err = ua.detectAppToken(ciMode, !isRecipe)
+
+	if err != nil {
+		return nil, nil, err
 	}
 
-	err := data.ValidateUploadToken(uploadToken)
+	//
+	// In CI mode, the target _must not_ be a recipe name:
```

Comment:
Isn't this a bit strict ?

If `waldo auth` worked for me on my computer my first instinct would be to use the exact same command on my CI. Why would it break ?

Reply:
This strictness comes from Larry the Enforcer. Talk to him as to his reasoning. I agree with your thinking.

Reply:
I'll convince it out of band XD